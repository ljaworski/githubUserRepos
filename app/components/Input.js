import React from "react";
import PropTypes from "prop-types";
import { Item, Input as NativeBaseInput } from "native-base";

export default function Input({ onChange, placeholder, value, ...props }) {
  return (
    <Item regular>
      <NativeBaseInput
        placeholder={placeholder}
        onChangeText={onChange}
        value={value}
        {...props}
      />
    </Item>
  );
}

Input.propTypes = {
  onChange: PropTypes.func,
  placeholder: PropTypes.string,
  value: PropTypes.string
};
