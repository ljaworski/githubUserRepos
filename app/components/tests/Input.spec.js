import React from "react";
import Input from "../Input";
import renderer from "react-test-renderer";

const props = {
  onChange: () => {},
  value: "test",
  placeholder: "Test"
};

test("renders correctly", () => {
  const input = renderer.create(<Input {...props} />).toJSON();
  expect(input).toMatchSnapshot();
});
