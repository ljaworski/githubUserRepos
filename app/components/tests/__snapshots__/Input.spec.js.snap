// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderBottomWidth": 1,
      "borderColor": "#D9D5DC",
      "borderLeftWidth": 1,
      "borderRadius": undefined,
      "borderRightWidth": 1,
      "borderTopWidth": 1,
      "borderWidth": undefined,
      "flexDirection": "row",
      "marginLeft": 2,
      "opacity": 1,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    editable={true}
    onChangeText={[Function]}
    placeholder="Test"
    placeholderTextColor="#575757"
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#000",
        "flex": 1,
        "fontSize": 17,
        "height": 50,
        "paddingLeft": 8,
        "paddingRight": 5,
        "top": 1.5,
      }
    }
    underlineColorAndroid="rgba(0,0,0,0)"
    value="test"
  />
</View>
`;
